src/index.ts

```diff
@@ -32,6 +105,43 @@ const plugin: PluginImpl<{}> = () => {
       };
     },
 
+    load(id) {
+      if (!tsx.test(id)) {
+        return null;
+      }
+      const source = program.getSourceFile(id);
+      if (!source) {
+        return null;
+      }
+      if (id.endsWith(dts)) {
+        return transformFile(source);
+      }
+      const ambientId = id.replace(tsx, dts);
```

Comment:
The `ambient` naming confuses me a bit, rather name it `definitionId` / `definitionSource`.

Reply:
or rather `declarationX`, to better reflect how its named in TS

Reply:
👍